### DISCUSSION TITLE
Existing collection nested subdocuments

### DISCUSSION THREAD

**Discussion by Mosquid** [typegoose/typegoose#736] [Q&A] [answered]
I have an existing collection of documents with an array of subdocuments. I'm trying to create a schema for this collection retroactively, e.g:

```ts
@modelOptions({
  schemaOptions: {
    _id: false,
    toJSON: { virtuals: true },
    toObject: { virtuals: true },
  },
})
class ChildClass {
  @prop()
  public description: string;
}

class ParentClass {
   @prop({ type: () => [ChildClass] })
   public children: ChildClass[];
}
```

`ParentClass.findOne` returns a doc without the "children" property as long as I `type: () => [ChildClass]`.  Changing it to `type: () => [Object]` does the trick but then I can't make the populate work...
`children` - is an array property of `ParentClass`, hence it's stored on the same document

**Comment by hasezoey** [comment, marked as answer]
What version are you using and what compiler with what options? also, could you maybe provide a sample document?

**Reply by Mosquid** [reply]
I'm using v9.3.1. 
Regarding the compiler, I assume you mean TypeScript? I'm using 4.3. I doubt ts version or tsconfig is the one to blame because i have a dozen perfectly functioning models.

Here's what my sample doc looks like:

```
{
  "_id": "62c6bc26ff87d90011ddffff",
  "createdBy": "61ed7d6aa6d0d600112ffff",
  "createdAt":"1657191462015",
  "projectName": "Sample Project",
  "wid": "62c6bc249daa437de55fffff",
  "publicURL": "https://example.com",
  "candidates": [
    {
      "rate": 1,
      "aid": "62c58d614be5aa0011xxxxx",
      "userId": "627bfa7a3538dd0011xxxxx",
      "role": "627bfa7a3538dd0011xxxxx"
    },
  ],
  "isShared": true,
  "sharedBy": "61ed7d6aa6d0d600112xxxx",
  "sharedAt": "1657191462015" 
}

```

**Reply by hasezoey** [reply]
weird, cannot reproduce, see https://github.com/typegoose/typegoose-testing/blob/verify736/src/test.ts

btw, minimal recommended typescript version by typegoose is 4.4

**Reply by Mosquid** [reply]
Oddly enough but I just went through all of the model properties and deleted them one by one. The problem was caused by one of the enum fields. So I got a new direction. I really appreciate your help

**Reply by Mosquid** [reply]
UPD: I had a property called `schema` and apparently it's a reserved name. Renaming this field solves the issue

**Reply by hasezoey** [reply]
in case it should help, a full list of reserved keys: https://github.com/Automattic/mongoose/blob/472601295c002d9865866c73e0d8f1f47916e995/lib/schema.js#L725-L783
